tests/scripts/generate_test_keys.py

```diff
@@ -157,6 +162,7 @@ def main() -> None:
 """)
     output_file.write("\n".join(look_up_table))
     output_file.write("\n};\n")
+    output_file.flush()
```

Comment:
Calling `flush` is only necessary if something is reading from the file while it's being generated. And that isn't happening. So why did you add it? This makes me worried that it's hiding some unrelated problem.

Reply:
I had to add this `flush` [recently](https://github.com/Mbed-TLS/mbedtls/pull/9017/commits/0ddab0ecee4642e13706328588056aaa2121c540) because the CI was failing complaining about the fact that the final look-up table in `test_keys.h` was not defined. I tried also it locally and indeed it was failing for the same reason (but the script itself was not failing). Adding the `flush` solved the problem so I kept it.

Reply:
Ok, so diving a bit deeper, the problem is that the script doesn't close the file explicitly. Closing the file does two things: it flushes the output buffer, and it throws an error if that last write can't be done (e.g. disk full).

I think flushing does a good enough job that the code is now correct. On the other hand, from what I've read ([1](https://realpython.com/why-close-file-python/#what-happens-if-you-dont-close-a-file-and-python-crashes) [2](https://stackoverflow.com/questions/17781906/what-happens-if-i-read-a-file-without-closing-it-afterwards)), flushing is necessary with some Python implementations, but not with CPython (where the file gets closed when `main` returns and `output_file` goes out of scope, and that throws an exception if the closure fails). And I think our CI only has CPython, so that wouldn't explain why adding `flush` was necessary to fix a practical problem. So I'm probably missing something, and I'm not satisfied with having `flush`.

Please use the [standard Python idiom](https://realpython.com/why-close-file-python/#toc) for working with a file:
```
with open(…) as output_file:
    output_file.write(…)
    …
```

(And if that indents the code too much, that's a sign that you should create an auxiliary function.)